experimental/sev_guest/src/instructions.rs

```diff
@@ -0,0 +1,193 @@
+//
+// Copyright 2022 The Project Oak Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+//! Rust instruction wrappers for managing page state and interacting with the hypervisor.
+
+use bitflags::bitflags;
+use core::arch::asm;
+use strum::FromRepr;
+
+/// Whether a page is in the validated state or not.
+#[derive(Debug, FromRepr)]
+#[repr(u32)]
+pub enum Validation {
+    /// The page is not validated.
+    Unvalidated = 0,
+    /// The page is validated.
+    Validated = 1,
+}
+
+/// The size of a memory page.
+#[derive(Debug, FromRepr)]
+#[repr(u32)]
+pub enum PageSize {
```

Comment:
Are 1G pages unsupported?

Reply:
No, not when running under SEV-ES or SEV-SNP.